spec/prog/postgres/postgres_server_nexus_spec.rb

```diff
@@ -100,142 +66,194 @@
+      Firewall.create(name: "#{aws_resource.ubid}-internal-firewall", location_id: aws_location.id, project_id: postgres_project.id)
+      postgres_timeline = PostgresTimeline.create
+
+      st = described_class.assemble(resource_id: aws_resource.id, timeline_id: postgres_timeline.id, timeline_access: "push", representative_at: Time.now)
+      postgres_server = PostgresServer[st.id]
+      # AWS uses AMI lookup via pg_ami method - verify the boot_image matches what pg_ami returns
+      expect(postgres_server.vm.boot_image).to eq(aws_location.pg_ami("17", "x64"))
     end
 
     it "raises error if the version is not supported for AWS" do
-      expect(PostgresResource).to receive(:[]).and_return(postgres_resource)
-      loc = Location.create(
-        name: "us-west-2",
-        display_name: "aws-us-west-2",
-        ui_name: "aws-us-west-2",
-        visible: true,
-        provider: "aws",
-        project_id: user_project.id
+      # Use an existing seed location that has billing rates but ensure no AMI mapping
+      aws_location = Location.find(name: "us-west-2")
+      skip "us-west-2 location not available in test seeds" unless aws_location
+
+      # Create resource with version 18 - ensure NO PgAwsAmi entry exists for this version
+      PgAwsAmi.where(aws_location_name: "us-west-2", pg_version: "18").each(&:destroy)
+
+      aws_resource = PostgresResource.create(
+        project_id: user_project.id,
+        location_id: aws_location.id,
+        name: "pg-aws-unsupported",
+        target_vm_size: "standard-2",
+        target_storage_size_gib: 64,
+        superuser_password: "dummy-password",
+        target_version: "18"
       )
-      expect(postgres_resource).to receive(:location).and_return(loc).at_least(:once)
-      expect(postgres_resource).to receive(:version).and_return("19").at_least(:once)
-      expect {
-        described_class.assemble(resource_id: postgres_resource.id, timeline_id: "91588cda-7122-4d6a-b01c-f33c30cb17d8", timeline_access: "push", representative_at: Time.now)
-      }.to raise_error NoMethodError, "undefined method 'aws_ami_id' for nil"
-    end
+      postgres_timeline = PostgresTimeline.create
 
-    it "errors out for unknown flavor" do
-      expect(PostgresResource).to receive(:[]).and_return(postgres_resource)
-      expect(postgres_resource).to receive(:flavor).and_return("boring_flavor").at_least(:once)
+      # Error happens when pg_ami returns nil for unsupported version
       expect {
-        described_class.assemble(resource_id: postgres_resource.id, timeline_id: "91588cda-7122-4d6a-b01c-f33c30cb17d8", timeline_access: "push", representative_at: Time.now)
-      }.to raise_error RuntimeError, "Unknown PostgreSQL flavor: boring_flavor"
+        described_class.assemble(resource_id: aws_resource.id, timeline_id: postgres_timeline.id, timeline_access: "push", representative_at: Time.now)
+      }.to raise_error NoMethodError, "undefined method 'aws_ami_id' for nil"
```

Comment:
Expecting code to raise a `NoMethodError` is odd. The prog should not be calling methods on an instance that may be nil. We should modify the prog not to do that. The `RuntimeError` raised previously was superior.